File: src/main/java/com/leet/easy/SymmetricTree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isSymmetricUtil(self, left: Optional[TreeNode], right: Optional[TreeNode]) -> bool:
        if left is None and right is None:
            return True
        if left is None or right is None:
            return False
        if left.val != right.val:
            return False
        if not self.isSymmetricUtil(left.right, right.left):
            return False
        if not self.isSymmetricUtil(left.left, right.right):
            return False
        return True

    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        return self.isSymmetricUtil(root.left, root.right)

File: src/main/java/com/leet/easy/SymmetricTree.exs
defmodule Solution do
  def solve(left, right) do
    cond do
      left == nil and right == nil -> true
      left == nil or right == nil -> false
      left.val != right.val -> false
      true -> solve(left.left, right.right) and solve(left.right, right.left)
    end
  end

  @spec is_symmetric(root :: TreeNode.t() | nil) :: boolean
  def is_symmetric(root) do
    solve(root.left, root.right)
  end
end
